### PR TITLE
GEODE-7864: Clean up InternalResourceManager

### PR DESCRIPTION
**Pull request by DonalEvans**
- Remove duplicate fine-level logging from deliverEventFromRemote()
 - Clean up raw use of parameterized classes
 - Clean up other static analyzer warnings

Authored-by: Donal Evans <doevans@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java [hunk 1]
```diff
@@ -125,99 +123,99 @@ public static InternalResourceManager createResourceManager(final InternalCache
...
     for (ResourceType resourceType : new ResourceType[] {ResourceType.HEAP_MEMORY,
         ResourceType.OFFHEAP_MEMORY}) {
-      Set<ResourceListener> emptySet = new CopyOnWriteArraySet<ResourceListener>();
-      this.listeners.put(resourceType, emptySet);
+      Set<ResourceListener<?>> emptySet = new CopyOnWriteArraySet<>();
+      listeners.put(resourceType, emptySet);
     }
 
-    this.closed = false;
+    closed = false;
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:164]: You got rid of "closed" being initialized to true so you should probably get rid of this line also. I think the old logic was: "start it at true and then at the end of the constructor set it to false (we must have had some code that allowed "this" to escape the constructor). We then set it to true in closed. But now its initial value is false so it does not make sense to reset it to false at the end of the constructor. We do pass "this" into the tempMonitors but those are just owned by this object so I don't see any real escape of "this".

Comment (DonalEvans) [reply geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:164]: Good catch, thanks. It seems that the only time we check the value of `closed` is to determine whether or not to log a warning after catching an exception in `runWithNotifyExecutor()`, so even in the event that the wrong value was used, the impact would be negligible.

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java [hunk 2]
```diff
@@ -228,41 +226,29 @@ public void removeResourceListener(final ResourceType resourceType,
   public void deliverEventFromRemote(final ResourceEvent event) {
     assert !event.isLocal();
```
Comment (aimiebell) [inline geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:225]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java [hunk 3]
```diff
@@ -125,99 +123,97 @@ public static InternalResourceManager createResourceManager(final InternalCache
...
     // Initialize the listener sets so that it only needs to be done once
     for (ResourceType resourceType : new ResourceType[] {ResourceType.HEAP_MEMORY,
         ResourceType.OFFHEAP_MEMORY}) {
-      Set<ResourceListener> emptySet = new CopyOnWriteArraySet<ResourceListener>();
-      this.listeners.put(resourceType, emptySet);
+      Set<ResourceListener<?>> emptySet = new CopyOnWriteArraySet<>();
+      listeners.put(resourceType, emptySet);
     }
-
-    this.closed = false;
```
Comment (mhansonp) [inline geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:166]: Why did this go away? It seems necessary off the cuff..

**Review by mhansonp — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java (hunk 3 above)
Comment (DonalEvans) [reply geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:166]: See the comment from Darrel above. We were previously initializing `closed` to true, then setting it to false at the end of the constructor. Now we simply initialize it to false and don't have to re-set it to false in the constructor.

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java [hunk 4]
```diff
@@ -228,41 +224,29 @@ public void removeResourceListener(final ResourceType resourceType,
   public void deliverEventFromRemote(final ResourceEvent event) {
     assert !event.isLocal();
 
-    if (this.cache.getLogger().fineEnabled()) {
-      this.cache.getLogger()
+    if (cache.getLogger().fineEnabled()) {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:227]: I would go ahead and replace all usages of the old proprietary LogWriter API (cache.getLogger() returns LogWriter) with Log4J2 Logger.

**Review by kirklund — APPROVED** [review]
Approved, but please consider updating from LogWriter to Logger whenever you change a class, especially for cleanup.

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java (hunk 4 above)
Comment (DonalEvans) [reply geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:227]: Will do!

File: geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java (hunk 3 above)
Comment (mhansonp) [reply geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java:166]: ok

**Review by mhansonp — APPROVED** [review]

**Review by nabarunnag — APPROVED** [review]
LGTM

**Pull request merged**